src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -46,3 +47,8 @@
 
   (stop [this context] context))
 
+(trapperkeeper/defservice jolokia-webservice
+  [[:WebserverService add-servlet-handler]]
+  (init [this context]
+    (add-servlet-handler (AgentServlet.) "/metrics/v2")
```

Comment:
I simplified things a bit by moving servlet handler into `defservice metrics-webservice` . This removes the need to add new service to `bootstrap.cfg`, which makes this a drop-in enhancement.

It also turns out that jolokia will defaults to looking for an access policy at `classpath:jolokia-access.xml`, so I tossed something under `resources/` that restricts access to the `write` and `exec` methods. I could see shipping this resource as a default and also allowing folks to configure a custom `jolokia-access.xml` file similar to what we do with `logback.xml`.

As for the getting a servlet into ring so that we could wrap middleware around it, I ran into the same roadblock mentioned earlier today: there seems to be documentation out in the community for turning a ring handler into a servlet, but not much for using a servlet as a ring handler.

Alternately, it looks like Jolokia has a stripped-down [HttpRequestHandler](https://github.com/rhuss/jolokia/blob/v1.3.4/agent/core/src/main/java/org/jolokia/http/HttpRequestHandler.java#L37-L43) that can consume raw GET and POST requests. This may be easier to wrap into a ring handler, but it looks like there isn't any automatic configuration of restrictions and other settings.